### ISSUE TITLE
ActionFailedException: Device failed to perform action: PAL_UPNP_SOAP_E_ACTION_FAILED

### ISSUE THREAD

**Issue by dotStart** [LordAkkarin/Beacon#22]
The application crashes due to an invalid router response.

**Reproduction Steps**
??? - Illegal response from router

**Expected Behavior**
An error message should appear

**Environment (please complete the following information):**
 - OS: Raspbian (Kernel Version: 5.4.79-v7l+)
 - Application Version: 3.0.0-alpha.3

**Additional Context**
```
Message: ActionFailedException: Device failed to perform action: PAL_UPNP_SOAP_E_ACTION_FAILED
  tv.dotstart.beacon.core.upnp.ActionExtensionsKt$invoke$2.invoke in ActionExtensions.kt::70
  tv.dotstart.beacon.core.upnp.ActionExtensionsKt$invoke$2.invoke in ActionExtensions.kt::-1
  net.mm2d.upnp.internal.impl.ActionImpl$invokeCustom$1$1.invoke in ActionImpl.kt::86
  net.mm2d.upnp.internal.impl.ActionImpl$invokeCustom$1$1.invoke in ActionImpl.kt::22
  net.mm2d.upnp.internal.thread.ExecuteFunctionKt$toRunnable$1.run in ExecuteFunction.kt::19
  java.util.concurrent.ThreadPoolExecutor.runWorker in ThreadPoolExecutor.java::1128
  java.util.concurrent.ThreadPoolExecutor$Worker.run in ThreadPoolExecutor.java::628
  java.lang.Thread.run in Thread.java::834
```

**Comment by dotStart** [comment]
Raygun report https://app.raygun.com/crashreporting/27ce4ol/errors/4810901503 has more details about the error

**Comment by dotStart** [comment]
The issue has most likely been resolved via the changes within b447e0dc48487eccb3fa582c083155bfdc248d6f but will remain open in the meantime as the stack trace does not seem to provide conclusive information regarding the source of the issue (as none of the mentioned frames would actually be able to throw an exception within the current version of the application).
